src/app/pages/OnDemandRadioPage/index.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useContext } from 'react';
+import { string, shape, object, arrayOf } from 'prop-types';
+import styled from 'styled-components';
+import path from 'ramda/src/path';
+import MetadataContainer from '../../containers/Metadata';
+import Grid, { GelPageGrid } from '#app/components/Grid';
+import RadioPageBlocks from '../../containers/RadioPageBlocks';
+import { ServiceContext } from '../../contexts/ServiceContext';
+
+const OnDemandRadioPage = ({ pageData }) => {
+  const blocks = path(['content', 'blocks'], pageData);
+  const promo = path(['promo'], pageData);
+  const metadata = path(['metadata'], pageData);
+  const { dir } = useContext(ServiceContext);
+  const StyledGelPageGrid = styled(GelPageGrid)`
+    flex-grow: 1;
+  `;
+  return (
+    <>
+      <MetadataContainer
+        title={promo.headlines.headline}
+        lang={metadata.language}
+        description={promo.media.synopses.short}
+        openGraphType="website"
+      />
+
+      <StyledGelPageGrid
+        forwardedAs="main"
+        role="main"
+        dir={dir}
+        columns={{
+          group0: 6,
+          group1: 6,
+          group2: 6,
+          group3: 6,
+          group4: 8,
+          group5: 20,
+        }}
+        enableGelGutters
+      >
+        <Grid
+          item
+          dir={dir}
+          startOffset={{
+            group0: 1,
+            group1: 1,
+            group2: 1,
+            group3: 1,
+            group4: 2,
+            group5: 5,
+          }}
+          columns={{
+            group0: 6,
+            group1: 6,
+            group2: 6,
+            group3: 6,
+            group4: 6,
+            group5: 12,
+          }}
+          margins={{ group0: true, group1: true, group2: true, group3: true }}
+        >
+          <RadioPageBlocks blocks={blocks} onDemand />
+        </Grid>
+      </StyledGelPageGrid>
+    </>
+  );
+};
+
+OnDemandRadioPage.propTypes = {
+  pageData: shape({
+    metadata: shape({
+      id: string,
+      tags: object,
+    }),
+    promo: shape({
+      subtype: string,
+      name: string,
+    }),
+    content: shape({
+      blocks: arrayOf(
+        shape({
+          uuid: string,
+          id: string,
+          externalId: string,
+          text: string,
+          type: string,
+        }),
+      ),
+    }),
+  }).isRequired,
+};
```

Comment:
Is it worth extracting this into a separate function like this: https://github.com/bbc/simorgh/blob/latest/src/app/models/propTypes/radioPage/index.js

Reply:
I've stressed so many times how important it is to keep prop types explicit and inline with the component and not to abstract them away because they it's super handy documentation for developers:

https://github.com/bbc/simorgh/issues/4477
https://gist.github.com/jroebu14/cb3f82f51d74a02a586e7bc430414a05
https://gist.github.com/jroebu14/0ff63ba198e7e46cfce2df3381710256

I'd prefer it if we kept them as is even if it means duplication. The problem is really that our prop type definitions are too complex because we pass components the full Ares response instead of picking out only the bits we need from the response before passing props down. An example of how I'd like our prop types to be https://github.com/bbc/simorgh/pull/5566/files#diff-e449f3ce3c0cbe3fd43e2629cf8661baR71-R76 rather having a hidden away centralised and hard to understand system of proptypes.

We should do this for all our components. I refactored the Metadata component a while ago to accept only the props it needs instead of the full Ares response and how look how simple it is to use now! https://github.com/bbc/simorgh/pull/5804/files#diff-6f9218412f92864ee8850958a05c23a2R21-R24

Sorry for all the reading but I'm passionate about this issue because it's about developer experience! Grab me if you want me to explain more.

Reply:
Excellent explanation - thank you. Maybe it would make sense to enventually delete the functions in /src/app/models/propTypes/ which are no longer needed or used anymore.  (But not in this PR obvs!)